### PR TITLE
Add toStreamGenerator (NonEmpty a) instance

### PR DESCRIPTION
**Pull request by phadej**
I'd like to have

```haskell
"stream" :> StreamGet NewlineFraming '[PlainText] [Text] -- lines
```

Also having `StreamingReqBody` should be in `servant`

### CODE REVIEW

File: servant-docs/src/Servant/Docs/Internal.hs [hunk 1]
```diff
@@ -840,6 +840,24 @@ instance OVERLAPPABLE_
           status = fromInteger $ natVal (Proxy :: Proxy status)
           p = Proxy :: Proxy a
 
+-- | TODO: mentiong it's streaming + framing strategy
```
Comment (alpmestan) [inline servant-docs/src/Servant/Docs/Internal.hs]: Typo + TODO :smile:

File: servant-docs/src/Servant/Docs/Internal.hs [hunk 2]
```diff
@@ -840,6 +840,24 @@ instance OVERLAPPABLE_
...
+-- Also there are no samples.
+instance OVERLAPPABLE_
+        (MimeRender ct a, KnownNat status
+        , ReflectMethod method)
+    => HasDocs (Stream method status framing ct a) where
+  docsFor Proxy (endpoint, action) DocOptions{..} =
+    single endpoint' action'
+
+    where endpoint' = endpoint & method .~ method'
+          action' = action &response.respTypes .~ allMime t
```
Comment (alpmestan) [inline servant-docs/src/Servant/Docs/Internal.hs]: spacing between & and response.

File: servant/src/Servant/API/Stream.hs [hunk 3]
```diff
@@ -41,24 +47,61 @@ type StreamGet  = Stream 'GET 200
...
+  buildFromStream x = runResultStream x $ \r -> do
+      e <- r
+      case e of
+          Nothing        -> fail "Empty stream"
+          Just (Left er) -> fail er
+          Just (Right y) -> do
+              ys <- lazyRead r
+              return (y :| ys)
+
+-- | TODO: hide from export
```
Comment (alpmestan) [inline servant/src/Servant/API/Stream.hs]: TODO to address

File: servant-docs/src/Servant/Docs/Internal.hs (hunk 1 above)
Comment (phadej) [reply servant-docs/src/Servant/Docs/Internal.hs]: Left intentionally, opened an issue already #989

**Pull request merged**